Add GA event label to Article, SignUp, and MO-clicks for event tracking

cl2205 opened the pull request:
@jonuy for your review :)

cl2205 committed:
Add GA event label to Article, SignUp, and MO-clicks for event tracking purposes

jonuy commented on js/dailyshine.js:
```diff
@@ -141,6 +141,7 @@
         if (i == messages.length - 1) {
           data.linkTitle = localized(content.linkTitle);
           data.linkUrl = localized(content.linkUrl);
+          data.dateClicked = formatDateMMddyyy(new Date());
```
Because people will have the ability to view content from previous days, instead of just using `new Date()` here, it should probably be something more like...

`getParameter('date') ? createDateFromQuery(getParameter('date')) : new Date()`

cl2205 replied:
Ahh, gotcha. For some reason I was thinking of the date when the user saw the Shine and clicked, as opposed to the date of the daily Shine, but it makes sense to capture the Shine’s date. Will fix this!

cl2205 committed:
- Fix GA event label to pass date of Shine content on article-click and sign-up-click
- Abstract more date logic into a separate getShineDate() helper function
- 'var' syntax error fix

cl2205 commented on js/dailyshine.js:
```diff
@@ -462,41 +463,58 @@
       // Used for the share link
       shareLink: 'http://' + window.location.hostname + '?date=' + dateQuery,
       showSignUp: getParameter('r') ? false : true,
+      shineDate: dateQuery
```
@jonuy: not sure if you intended the date for sign-up click to be treated any differently from article-click -- I was also initially thinking it would be the date the user hit 'Sign up' (in which case I think it would be the current day?), but here I've also changed it to the date of that particular Shine content that motivated her to sign up.  Let me know if this is not the case :)

jonuy replied:
Ah I see where the confusion about that came from. Sorry about that! GA already should be tracking itself the time an event occurs, so yea, this is more about what content is it that prompted the sign up.

jonuy reviewed (approved):
Looks good! Thanks 👍
